### PR TITLE
Change workspace docker/directory names to be unique

### PR DESCRIPTION
**Pull request by tdy**
* [x] change to unique container/directory `localName=workspace-<UUID>` mapped to `s3Name=workspace-<ID>`
* [x] fix minor unrelated logging error in `lib/workspace.js`

### CODE REVIEW

File: workspace_host/interface.js [hunk 1]
```diff
@@ -335,17 +336,48 @@ function _recursiveUploadJobManager(curDirPath, S3curDirPath) {
...
+        logger.info(`watch: workspace_id=${workspace_id}, localPath=${localPath}`);
+        let s3Path;
+        if (!workspace_id) {
+            logger.info(`watch return: workspace_id not mapped yet`);
+            return;
+        } else if (localPath === '') {
+            logger.info(`watch continue: empty (root) path`);
+            continue;
+        } else {
+            s3Path = `workspace-${workspace_id}/${localPath}`;
```
Comment (mwest1066) [inline workspace_host/interface.js:370]: Should this be using `s3Name`?

Comment (tdy) [reply workspace_host/interface.js:370]: I think there's a race condition where fswatcher instantly sees the dirs but `s3Name` might not have been assigned yet by the workspace initializer.

Comment (mwest1066) [reply workspace_host/interface.js:370]: As we discussed in the meeting, I think we should be checking that the `id_workspace_mapper` is correctly initialized for this workspace and returning an error if it isn't. An error here probably means there is an old leftover workspace still running that we should be ignoring.

Comment (tdy) [reply workspace_host/interface.js]: Handled in `_getWorkspaceByPath()` now:

https://github.com/PrairieLearn/PrairieLearn/pull/2777/files#diff-e7517976b89db12af19f23b917645289R422-R427

**Pull request merged**